pkg/ui/bucket.go

```diff
@@ -30,13 +30,13 @@ func NewBucketUI(logger log.Logger, label string) *Bucket {
 }
 
 // Register registers http routes for bucket UI.
-func (b *Bucket) Register(r *route.Router, ins extpromhttp.InstrumentationMiddleware) {
+func (b *Bucket) Register(s server.Server, ins extpromhttp.InstrumentationMiddleware) {
 	instrf := func(name string, next func(w http.ResponseWriter, r *http.Request)) http.HandlerFunc {
 		return ins.NewHandler(name, http.HandlerFunc(next))
 	}
 
-	r.Get("/", instrf("root", b.root))
-	r.Get("/static/*filepath", instrf("static", b.serveStaticAsset))
+	s.Handle("/", instrf("root", b.root))
+	s.Handle("/static/*filepath", instrf("static", b.serveStaticAsset))
```

Comment:
Why change these specifically?

Reply:
:/ because the `Server` type doesn't have a `Get` method, only a `Handle` and to be consistent internally, it simplifies things to re-use the server package.

Reply:
`Server` is just a wrapper around `http.Server` that why it resonates with its methods. However, since `prometheus/common/route.Router` implements `http.Handler` you can pass it down to the `Server`, if you want to preserve behaviour.

Reply:
I guess we could do that but I think it would just add an extra layer of complexity